### PR TITLE
clp: Add maximum length for utf8 validation (fixes #242).

### LINKED ISSUE #242: Compression error on var-logs
### Bug

The CLP compressor runs into an error when compressing our internal var-log.  some example error messages are attached below

```
2024-01-24 00:57:45,022 [error] Cannot compress hostd-10 - not an IR stream or UTF-8 encoded
2024-01-24 00:57:45,125 [error] Cannot compress hostd-12 - not an IR stream or UTF-8 encoded
```

Note that 4f9ed0abfe3769910e7d57ad45da0a1fe8831ffe doesn't have this issue, but we don't know from which commit did the issue start

### CLP version

5d6ff54d375af35f8aadd7d551672c255f627ab9

### Environment

ubuntu 22.04

### Reproduction steps

./clp c varlog_compressed /<path>/<to>/var-logs

### PR DESCRIPTION
**Pull request by haiqi96**
# References
https://github.com/y-scope/clp/issues/242

# Description
The issue identified in https://github.com/y-scope/clp/issues/242 is not a real issue. It was caused by a size change in the validation buffer. so a non-utf8 char that was not detected by the previous 4k buffer size is revealed by 64k buffer. The decision is to let CLP only read the first 4096 bytes of a file for utf8 validation.

# Validation performed
Compressed the var-logs and verified that the error observed in issue 242 is gone.

### CODE REVIEW

File: components/core/src/clp/clp/FileCompressor.hpp [hunk 1]
```diff
@@ -146,6 +146,7 @@ class FileCompressor {
     );
 
     // Variables
+    size_t const m_utf_max_validation_len = 4096;
```
Comment (LinZhihao-723) [inline components/core/src/clp/clp/FileCompressor.hpp:149]: Maybe use `constexpr`?

File: components/core/src/clp/clp/FileCompressor.hpp [hunk 2]
```diff
@@ -50,6 +50,9 @@ class FileCompressor {
     );
 
 private:
+    // Constants
+    static constexpr size_t m_utf_max_validation_len = 4;
```
Comment (LinZhihao-723) [inline components/core/src/clp/clp/FileCompressor.hpp:54]:
```suggestion
    static constexpr size_t cUtfMaxValidationLen = 4;
```
naming convention for constexpr data members

File: components/core/src/clp/clp/FileCompressor.cpp [hunk 3]
```diff
@@ -142,9 +142,10 @@ bool FileCompressor::compress_file(
         return false;
     }
     char const* utf8_validation_buf{nullptr};
-    size_t utf8_validation_buf_len{0};
-    m_file_reader.peek_buffered_data(utf8_validation_buf, utf8_validation_buf_len);
+    size_t peek_size{0};
+    m_file_reader.peek_buffered_data(utf8_validation_buf, peek_size);
     bool succeeded = true;
+    size_t utf8_validation_buf_len = std::min(peek_size, m_utf_max_validation_len);
```
Comment (kirkrodrigues) [inline components/core/src/clp/clp/FileCompressor.cpp:148]:
```suggestion
    auto utf8_validation_buf_len = std::min(peek_size, m_utf_max_validation_len);
```

File: components/core/src/clp/clp/FileCompressor.cpp [hunk 4]
```diff
@@ -355,9 +356,10 @@ bool FileCompressor::try_compressing_as_archive(
             continue;
         }
         char const* utf8_validation_buf{nullptr};
-        size_t utf8_validation_buf_len{0};
-        m_libarchive_file_reader.peek_buffered_data(utf8_validation_buf, utf8_validation_buf_len);
+        size_t peek_size{0};
+        m_libarchive_file_reader.peek_buffered_data(utf8_validation_buf, peek_size);
         string file_path{m_libarchive_reader.get_path()};
+        size_t utf8_validation_buf_len = std::min(peek_size, m_utf_max_validation_len);
```
Comment (kirkrodrigues) [inline components/core/src/clp/clp/FileCompressor.cpp:362]:
```suggestion
        auto utf8_validation_buf_len = std::min(peek_size, m_utf_max_validation_len);
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]
Small changes

File: components/core/src/clp/clp/FileCompressor.hpp [hunk 5]
```diff
@@ -144,7 +147,6 @@ class FileCompressor {
             streaming_archive::writer::Archive& archive,
             ir::LogEventDeserializer<encoded_variable_t>& log_event_deserializer
     );
-
```
[the remark this answers is not in the document]
Comment (haiqi96) [reply components/core/src/clp/clp/FileCompressor.hpp]: GG, must have caused by very slow clion when I made the 2nd fix

**Review by kirkrodrigues — APPROVED** [review]

**Review by LinZhihao-723 — APPROVED** [review]
lgtm

**Pull request merged**